lib/features/drilldown/DrilldownBreadcrumbs.js

```diff
@@ -21,15 +21,28 @@ export default function DrilldownBreadcrumbs(eventBus, elementRegistry, overlays
   var containerClasses = classes(container);
   container.appendChild(breadcrumbs);
 
-  function updateBreadcrumbs(element) {
-    var parents = getParentChain(element);
+  function updateBreadcrumbs(currentRoot) {
+    var parents = getParentChain(currentRoot);
 
-    var path = parents.map(function(el) {
-      var title = escapeHTML(el.name || el.id);
+    var path = parents.map(function(parent) {
+      var title = escapeHTML(parent.name || parent.id);
       var link = domify('<li><span class="bjs-crumb"><a title="' + title + '">' + title + '</a></span></li>');
 
+      var parentPlane = canvas.findRoot(planeId(parent)) || canvas.findRoot(parent.id);
+
+      if (!parentPlane && is(parent, 'bpmn:Process')) {
+
+        // root is a collaboration
+        var participant = elementRegistry.find(function(element) {
```

Comment:
Instead of searching through all diagram elements like this, can't we search through all roots to find the element (participant _or_ process backed element) that references the diagram? It is a difference between searching through 5 or 500 elements, every time we update.

Also, I'd personally expect the magic to happen in `getParentChain`, not in the code boostrapping the HTML. But that is just a personal preference.

Reply:
> can't we search through all roots
Fair point, I will adjust it 👍

> I'd personally expect the magic to happen in `getParentChain`
I'd argue that `getParentChain` gives us the execution semantics and the actual linking to the corresponding planes should happen somewhere else

Reply:
As both are personal preference I#d love to get your feedback. Happy to merge things as is (and see how they evolve over time, too).

We could add a comment like https://github.com/bpmn-io/bpmn-js/pull/1567#issuecomment-1011361926 or make the variables easier to grasp (i.e. rename parent to `parentBo`) to outline which problem this piece of code solves.

Reply:
Maybe `parentBoChain` will make it clearer that we are dealing with parent hierarchy entirely on the business object level here (and must compensate for the case where the business object does not have a direct graphical represnetation, but one linked through the displayed participant).

Reply:
> can't we search through all roots

On second thought, the root element is the Collaboration, not the Participant, and does not have a direct link to the Processes. We also have to check that it is the correct Collaboration in Case we have a multi-Collaboration-Diagram.

So the logic here would be more complex if we went for root elements. As this is only done once every drilldown and not every render, let's keep the logic "simple", even when it's less efficient.

Reply:
I'll look into explaining the magic in the code tomorrow (:

Reply:
Added documentation with dfe2f8eb5ffd204faa1b6808b250ec48470a39c6

Reply:
:tada: